Kernel/Net/IPv4Socket.cpp

```diff
@@ -169,12 +169,12 @@ ssize_t IPv4Socket::sendto(FileDescription&, const void* data, size_t data_lengt
         m_peer_port = ntohs(ia.sin_port);
     }
 
-    auto adapter = adapter_for_route_to(m_peer_address);
-    if (!adapter)
+    auto routing_decision = route_to(m_peer_address, m_local_address);
+    if (!routing_decision.adapter || routing_decision.next_hop.is_zero())
```

Comment:
Instead of checking both of these every time we get a `RoutingDecision`, maybe we could add a convenient bool getter member that checks both?

Reply:
Seems like a nice ergonomic improvement - will do.